docs/integrations/web-servers/heroku.md

```diff
@@ -0,0 +1,359 @@
+
+### Sample log message
+
+* A Logplex POST body resembles the following:
+```
+83 <40>1 2012-11-30T06:45:29+00:00 host app web.3 - State changed from starting to up
+119 <40>1 2012-11-30T06:45:26+00:00 host app web.3 - Starting process with command `bundle exec rackup config.ru -p 24405
+```
+
+* Runtime metric logs have the following format:
+```
+335 <134>1 2023-08-24T10:28:47.153192+00:00 host heroku web.1 - source=web.1 dyno=heroku.322071457.63c5abfd-838b-4e2d-bce1-ce46de280675 sample#memory_total=180.43MB sample#memory_rss=180.05MB sample#memory_cache=0.38MB sample#memory_swap=0.00MB sample#memory_pgpgin=84329pages sample#memory_pgpgout=38140pages sample#memory_quota=512.00MB
+205 <134>1 2023-08-24T12:31:50.112+00:00 host heroku web.1 - source=web.1 dyno=heroku.319324155.67cc34d0-0440-4106-97b6-d9486f7d9009 sample#load_avg_1m=0.00 sample#load_avg_5m=0.00 sample#load_avg_15m=0.01
+```
+
+### Sample query
+
+#### Log Query
+
+* **Successfull App Build Trend** panel query:
+```sql
+_sourceCategory="Heroku" "Build Succeeded"
+| where _sourceName matches "{{log_drain}}"
+| _sourceName as log_drain
+| timeslice 5m
+| count by log_drain, _timeslice
+| transpose row _timeslice column log_drain
+```
+
+#### Metric Log Query
+
+* **Memory Utilization (MB)** panel query:
+```sql
+_sourceCategory="Heroku"
+| parse regex "dyno=(?<dyno>.*?(?= )).*memory_total=(?<memory_total>.*?(?=MB )).*memory_rss=(?<memory_rss>.*?(?=MB )).*memory_cache=(?<memory_cache>.*?(?=MB )).*memory_swap=(?<memory_swap>.*?(?=MB ))"
+| where dyno matches "{{dyno}}" and _sourceName matches "{{log_drain}}"
+| timeslice 1m
+| avg(memory_total) as memory_total, avg(memory_rss) as resident_memory, avg(memory_cache) as disk_cache_memory, avg(memory_swap) as swap_memory by _timeslice
+```
+
+## Collecting Logs for Heroku
+
+Heroku is a cloud platform that lets companies build, deliver, monitor and scale apps in 8 programming languages namely Node.js, Ruby, Python, Java, PHP, Go, Scala and Clojure.
+
+There are two ways to send Heroku logs to Sumo:
+* [Sumo Logic Add-on](https://elements.heroku.com/addons/sumologic) : The Sumo Logic Add-on can be attached to a Heroku application via the the CLI or the UI. It automatically creates a Sumo Logic free trial account that contains a [https logs source](https://help.sumologic.com/docs/send-data/hosted-collectors/http-source/logs-metrics/) on a [hosted collector](https://help.sumologic.com/docs/send-data/hosted-collectors/configure-hosted-collector/).
+* [HTTPS Log Drain](https://devcenter.heroku.com/articles/log-drains#https-drains) : A HTTPS Log Drain can be attached to a Heroku application via the CLI. It can be used to send logs to a [http logs source](https://help.sumologic.com/docs/send-data/hosted-collectors/http-source/logs-metrics/) for any type of [Sumo Logic account](https://help.sumologic.com/docs/manage/manage-subscription/) as per your requirement. The Sumo Logic Add-on internally wraps this method.
```

Comment:
```suggestion
* **[HTTPS Log Drain](https://devcenter.heroku.com/articles/log-drains#https-drains)**. A HTTPS Log Drain can be attached to a Heroku application via the CLI. It can be used to send logs to a [http logs source](/docs/send-data/hosted-collectors/http-source/logs-metrics/) for any type of [Sumo Logic account](/docs/manage/manage-subscription/) as per your requirement. The Sumo Logic Add-on internally wraps this method.
```